### PR TITLE
2411 with thrift changes

### PR DESCRIPTION
**Pull request by dlmarion**
This change cherry-picks 2960311 from #2620 and builds upon it to split the TabletClientService into two Thrift services, one for scan and one for write (or the remainder of the current API).

### CODE REVIEW

**Comment by dlmarion** [comment]
Cherry-picked e243d49, c0814db, and 47ef16d from https://github.com/apache/accumulo/pull/2620

**Comment by dlmarion** [comment]
Cherry-picked d582824 and 5df51dd from apache#2620

File: server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+import com.google.common.collect.Collections2;
+
+public class ThriftScanClientHandler implements TabletScanClientService.Iface {
+
+  private static final Logger log = LoggerFactory.getLogger(ThriftScanClientHandler.class);
+
+  private final TabletServer server;
+  protected final ServerContext context;
+  protected final SecurityOperation security;
+  private final WriteTracker writeTracker = new WriteTracker();
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java]: @dlmarion  in the tserver need to share this with the TabletClientHandler, like the tablet transaction watcher

**Comment by keith-turner** [comment]
Since I had already reviewed apache/accumulo#2620 I didn't want to completely review this since a lot is redundant.  I tried merging main into this branch so I could more easily see the diffs that are just for the scan server.  I did that merge in the following branch, however I don't feel I got everything correct when resolving conflicts.   Posting my attempt in case you want to look at it, but be careful if you use it.

https://github.com/keith-turner/accumulo/tree/2411_with_thrift_changes

I did find that once problem w/ writeTracker when attempting to review these changes before switching gears to merging inorder to simplify the review.  Now that I finished the merge I am going to sign off for the day and continue the review tomorrow.

**Comment by keith-turner** [comment]
This is my merge commit in that branch I posted in the last comment.

https://github.com/keith-turner/accumulo/commit/9b128932fd1051bc0d0515e1de05248f0b35c6d1

**Comment by dlmarion** [comment]
Did you happen to run the ITs? I'm in the process of merging `main` into the main feature branch and the ITs are failing. I'm working on resolving that first, then will look at merging the main feature branch into this branch.

**Comment by dlmarion** [comment]
I figured it out, I made a mistake in the merge from `main`.

**Pull request merged**